pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultiStageOperator.java

```diff
@@ -50,6 +64,17 @@ public TransferableBlock nextBlock() {
       // TODO: move this to centralized reporting in broker
       if (nextBlock.isEndOfStreamBlock()) {
         LOGGER.info("Recorded operator stats: " + _operatorStats);
+        if (nextBlock.isSuccessfulEndOfStreamBlock()) {
+          for (MultiStageOperator op : getChildOperators()) {
+            _operatorStatsMap.putAll(op.getOperatorStatsMap());
+          }
+
+          if (!_operatorStats.getExecutionStats().isEmpty()) {
+            _operatorStatsMap.put(toExplainString() + "_" + _requestId + "_" + _stageId, _operatorStats);
```

Comment:
here we should include explainName, stageId, requestId, and the serverId so that upstream operators putAll will not overwrite one server's metadata on top of another.